src/rt-app.c

```diff
@@ -590,44 +590,39 @@ static void set_thread_affinity(thread_data_t *data, cpuset_data_t *cpu_data)
 	}
 }
 
-void *thread_body(void *arg)
+static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
 {
-	thread_data_t *data = (thread_data_t*) arg;
-	phase_data_t *pdata;
-	log_data_t ldata;
 	struct sched_param param;
-	struct timespec t_start, t_end, t_first;
-	unsigned long t_start_usec;
-	long slack;
-	timing_point_t *curr_timing;
-	timing_point_t *timings;
-	timing_point_t tmp_timing;
-	unsigned int timings_size, timing_loop;
+	policy_t policy;
+#ifdef DLSCHED
+	struct sched_attr dl_params;
 	pid_t tid;
-	struct sched_attr attr;
 	unsigned int flags = 0;
-	int ret, phase, phase_loop, thread_loop, log_idx;
+#endif
+	int ret;
 
-	/* Set thread name */
-	ret = pthread_setname_np(pthread_self(), data->name);
-	if (ret !=  0) {
-		perror("pthread_setname_np thread name over 16 characters");
+	if (sched_data == NULL)
```

Comment:
This is what is done for cpuset but i don't really like it for priority because with such behavior, you must add scheduler properties in every phase object even if you don't change compared to the previous phase. This behavior is more like other events in the phase for changing scheduler properties